src/profile-logic/marker-data.js

```diff
@@ -1159,6 +1160,18 @@ export function isFileIoMarker(marker: Marker): boolean {
   return !!(marker.data && marker.data.type === 'FileIO');
 }
 
+/**
+ * Returns true if the marker is an on-thread FileIO marker.
```

Comment:
can you please elaborate a bit more in the comment? What are the different "types" of  FileIO markers?

Reply:
(Responding to some questions that I know from the back-end size; Nazim please correct me if I'm wrong on the front-end side.)
Technically there is only one "type" of "FileIO" markers (those where data.type=="FileIO"), but we will distinguish those that originated on the current thread (data.threadId===undefined), and those that happened on another thread but were recorded on this thread (data.threadId is the thread number where the IO happened).
The latter kind will be used to record IOs from all threads into the main thread, so perf people interested in IOs can find them all in one track.

Reply:
Thanks for the comment Gerald, added more comments inside the code.